### PR TITLE
chore(cmake): do not cache git version

### PR DESCRIPTION
**Pull request by GMishx**
## Description

1. Do not cache git version information in cmake, allowing rebuild of VERSION files at each commit change using `cmake --build` command.
2. Clean TWIG cache while installing `www` packages preventing confusion while working on UI.

### Changes

1. Remove git variables from cmake cache.
2. Add new install target in `www` to clean twig cache.

## How to test

1. Change git commit and run `cmake --build`, the VERSION files do not update showing old version information in UI.
2. With the commit, every change in git commit should cause rebuild of VERSION files.
3. At each install, the twig cache should get cleaned (`/var/local/cache/fossology`)

<a href="https://gitpod.io/#https://github.com/fossology/fossology/pull/2362"></a>

### CODE REVIEW

**Comment by GMishx** [comment]
@avinal , can you please review the changes?

File: cmake/SetDefaults.cmake [hunk 1]
```diff
@@ -116,19 +116,30 @@ if(DEFINED CMAKE_CXX_COMPILER)
     find_package(Boost REQUIRED regex system filesystem program_options)
 endif()
 find_package(Git REQUIRED)
-foreach(SCHE_LIBS glib-2.0 gthread-2.0 gio-2.0 gobject-2.0)
+
+# libmagic, libgcrypt does not have pc module on Debian buster
+# json-c does not have cmake aware packages on Debian buster
+foreach(SCHE_LIBS
+        glib-2.0 gthread-2.0 gio-2.0 gobject-2.0 rpm libxml-2.0 libxslt icu-uc
+        json-c
```
Comment (avinal) [inline cmake/SetDefaults.cmake:124]: Two packages have been removed but only one was added back, is the other one not required?

File: CMakeLists.txt [hunk 2]
```diff
@@ -9,6 +9,13 @@ project(FOSSOLOGY)
 # set defaults for the project
 include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/SetDefaults.cmake)
 
+# Check latest version info from git
+message(STATUS "Current fossology version ${FO_VERSION}")
+message(STATUS "Latest commit hash: ${FO_COMMIT_HASH}")
+message(STATUS "Current Branch: ${FO_BRANCH}")
+message(STATUS "Current Build Data: ${FO_BUILD_DATE}")
```
Comment (avinal) [inline CMakeLists.txt:16]:
```suggestion
message(STATUS "Current Build Date: ${FO_BUILD_DATE}")
```
small typo

File: cmake/FoUtilities.cmake [hunk 3]
```diff
@@ -81,7 +77,6 @@ macro(getGitVersion)
         WORKING_DIRECTORY ${FO_SOURCEDIR}
     )
     set(FO_COMMIT_DATE ${COMMIT_DATE} CACHE INTERNAL "latest commit hash")
```
Comment (avinal) [inline cmake/FoUtilities.cmake:79]:
```suggestion
    set(FO_COMMIT_DATE ${COMMIT_DATE} CACHE INTERNAL "latest commit date")
```

File: cmake/SetDefaults.cmake [hunk 4]
```diff
@@ -116,19 +116,30 @@ if(DEFINED CMAKE_CXX_COMPILER)
...
 endforeach()
-pkg_check_modules(jsoncpp REQUIRED jsoncpp)
-pkg_check_modules(jsonc REQUIRED json-c)
+
+if(TESTING)
+    foreach(TEST_LIBS
+            cunit cppunit
+    )
+        string(REPLACE "-2.0" "" LIB_NAME ${TEST_LIBS})
+        pkg_check_modules(${LIB_NAME} REQUIRED ${TEST_LIBS})
```
Comment (avinal) [inline cmake/SetDefaults.cmake:135]:
```suggestion
        pkg_check_modules(${TEST_LIBS} REQUIRED ${TEST_LIBS})
```
Nothing to be replaced.

**Review by avinal — CHANGES_REQUESTED** [review]
Hi @GMishx, I have a few suggestions, the rest looks okay.

File: CMakeLists.txt (hunk 2 above)
Comment (GMishx) [reply CMakeLists.txt]: Done.

File: cmake/FoUtilities.cmake (hunk 3 above)
Comment (GMishx) [reply cmake/FoUtilities.cmake]: Done.

File: cmake/SetDefaults.cmake (hunk 4 above)
Comment (GMishx) [reply cmake/SetDefaults.cmake]: Done.

File: cmake/SetDefaults.cmake (hunk 1 above)
Comment (GMishx) [reply cmake/SetDefaults.cmake:124]: `jsoncpp` has been added to copyright package as currently it only uses it.

**Review by shaheemazmalmmd — APPROVED** [review]
Code looks Good. tested works as expected.

**Pull request merged**